tools/restatectl/src/commands/cluster/provision.rs

```diff
@@ -59,11 +56,9 @@ async fn cluster_provision(
     connection_info: &ConnectionInfo,
     provision_opts: &ProvisionOpts,
 ) -> anyhow::Result<()> {
-    let node_address = provision_opts
-        .address
-        .clone()
-        .unwrap_or_else(|| connection_info.cluster_controller.clone());
-    let channel = grpc_channel(node_address.clone());
+    let Some((_, channel)) = connection_info.channels().next() else {
+        anyhow::bail!("At least one node address is required");
+    };
```

Comment:
What we have to make sure for the cluster-provision command is that the call is idempotent. So if the user retries, then it shouldn't result into another node being provisioned (unless he explicitly ask us to do so). Accidentally provisioning multiple nodes is a very tricky situation to be in. Moreover, if the user is using the embedded metadata store, then he needs to hit a node that has the `metadata-server` role. If he uses an external metadata store, then he can hit any node.

Reply:
@tillrohrmann I understand. This is why I am only taking the first address provided by the command line. This should be idempotent unless the user changed the params they are providing to the command line.

An improvement maybe to fail if multiple addresses are given and ask the user to only pass one and to stick to it maybe?

Reply:
I'm not quite following this discussion - this calls into `NodeCtlSvcHandler::provision_cluster`, from where we make a series of updates to cluster metadata using the metadata client configuration on the node on which we land. The updates themselves are conditional updates to versioned metadata values, and we appear to cater for retries. In particular, writing the initial nodes config is conditional on `Precondition::DoesNotExist`. What am I missing?

Reply:
I think @tillrohrmann should answer that. But what I think the problem is that if you provision 2 different nodes, you can end up with 2 split clusters of nodes that has different leaders. Beside the fact that you will have completely different version 1 of the metadata (and probably other versions for each new node that joins) I don't believe the raft protocol can eventually converse into a consistent state after this initial split brain.

Reply:
Azmy is right. The problem we want to avoid is that two different nodes get provisioned and start different metadata clusters so that the cluster segregates into smaller clusters.